presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -311,6 +311,36 @@ public void testGroupByWithSubquerySelectExpressionWithDereferenceExpression()
     }
 
     @Test
+    public void testGroupByWithSubquerySelectExpressionWithLambda()
+    {
+        // GROUP BY column captured in lambda
+        analyze(
+                "SELECT (SELECT apply(0, x -> x + b) from (values 1) x(a)) " +
+                        "FROM t1 u GROUP BY b");
+
+        // non-GROUP BY column captured in lambda
+        assertFails(
```

Comment:
please also add tests with lambdas with capture in subquery in ORDER BY expression. The capture should reference:
- positive: output column
- positive: grouping column (expressions are not supported, right)
- negative: non grouping columns